src/audio/google_rtc_audio_processing.c

```diff
@@ -0,0 +1,387 @@
+}
+
+void GoogleRtcFree(void *ptr)
+{
+	return rfree(ptr);
+}
+
+static int google_rtc_audio_processing_params(
+	struct comp_dev *dev,
+	struct sof_ipc_stream_params *params)
+{
+	int ret;
+
+	ret = comp_verify_params(dev, 0, params);
+	if (ret < 0) {
+		comp_err(
+			dev,
+			"google_rtc_audio_processing_params(): comp_verify_params failed.");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int google_rtc_audio_processing_cmd_set_data(
+	struct comp_dev *dev,
+	struct sof_ipc_ctrl_data *cdata)
+{
+	struct google_rtc_audio_processing_comp_data *cd = comp_get_drvdata(dev);
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_BINARY:
+		return comp_data_blob_set_cmd(cd->tuning_handler, cdata);
+	default:
+		comp_err(dev,
+				 "google_rtc_audio_processing_ctrl_set_data(): Only binary controls "
+				 "supported %d",
+				 cdata->cmd);
+		return -EINVAL;
+	}
+}
+
+static int google_rtc_audio_processing_cmd_get_data(
+	struct comp_dev *dev,
+	struct sof_ipc_ctrl_data *cdata,
+	int max_data_size)
+{
+	comp_info(dev, "google_rtc_audio_processing_ctrl_get_data(): %u", cdata->cmd);
+	struct google_rtc_audio_processing_comp_data *cd = comp_get_drvdata(dev);
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_BINARY:
+		return comp_data_blob_get_cmd(cd->tuning_handler, cdata, max_data_size);
+	default:
+		comp_err(dev,
+				 "google_rtc_audio_processing_ctrl_get_data(): Only binary controls "
+				 "supported %d",
+				 cdata->cmd);
+		return -EINVAL;
+	}
+}
+
+static int google_rtc_audio_processing_cmd(struct comp_dev *dev, int cmd, void *data,
+									 int max_data_size)
+{
+	struct sof_ipc_ctrl_data *cdata = data;
+
+	comp_dbg(dev, "google_rtc_audio_processing_cmd(): %d - data_cmd: ", cmd, cdata->cmd);
+
+	switch (cmd) {
+	case COMP_CMD_SET_DATA:
+		return google_rtc_audio_processing_cmd_set_data(dev, cdata);
+	case COMP_CMD_GET_DATA:
+		return google_rtc_audio_processing_cmd_get_data(dev, cdata, max_data_size);
+	default:
+		comp_err(dev, "google_rtc_audio_processing_cmd(): Unknown cmd %d", cmd);
+		return -EINVAL;
+	}
+}
+
+static struct comp_dev *google_rtc_audio_processing_create(
+	const struct comp_driver *drv,
+	struct sof_ipc_comp *comp_template)
+{
+	struct comp_dev *dev = NULL;
+	struct sof_ipc_comp_process *comp;
+	struct google_rtc_audio_processing_comp_data *cd = NULL;
+	int ret;
+
+	comp_cl_info(drv, "google_rtc_audio_processing_create()");
+
+	/* Create component device with an effect processing component */
+	dev = comp_alloc(drv, COMP_SIZE(struct sof_ipc_comp_process));
+	if (!dev)
+		return NULL;
+
+	/* Create private component data */
+	cd = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, sizeof(*cd));
+	if (!cd)
+		goto fail;
+
+	comp = COMP_GET_IPC(dev, sof_ipc_comp_process);
+	ret = memcpy_s(comp, sizeof(*comp), comp_template,
+				   sizeof(struct sof_ipc_comp_process));
+	assert(!ret);
+
+	cd->tuning_handler = comp_data_blob_handler_new(dev);
+	if (!cd->tuning_handler)
+		goto fail;
+
+	cd->state = GoogleRtcAudioProcessingCreate();
```

Comment:
there are no error logs here but a lot of error paths, my be worth adding some for the external lib calls at a minimum